internal/validation/v2alpha1/gateway.go

```diff
@@ -2,20 +2,24 @@ package v2alpha1
 
 import (
 	"fmt"
+
 	gatewayv2alpha1 "github.com/kyma-project/api-gateway/apis/gateway/v2alpha1"
 	"github.com/kyma-project/api-gateway/internal/validation"
 	networkingv1beta1 "istio.io/client-go/pkg/apis/networking/v1beta1"
 )
 
 func validateGateway(parentAttributePath string, gwList networkingv1beta1.GatewayList, apiRule *gatewayv2alpha1.APIRule) []validation.Failure {
 	var failures []validation.Failure
-	gatewayName := *apiRule.Spec.Gateway
 
-	if !gatewayExists(gwList, gatewayName) {
-		failures = append(failures, validation.Failure{
-			AttributePath: parentAttributePath + ".gateway",
-			Message:       "Gateway not found",
-		})
+	if apiRule.Spec.Gateway != nil {
```

Comment:
It should already fail here, not return empty

Reply:
let me check if this is applicable, for APIRules that can not be converted (e.g. empty spec).

Reply:
looks good, added additional/more general validation step for empty spec